ergo/distributions/point_density.py

```diff
@@ -46,16 +45,31 @@ def __init__(
             self.normed_xs = scale.normalize_points(xs)
             self.normed_densities = scale.normalize_densities(self.normed_xs, densities)
 
-        self.bin_probs = self.normed_densities * constants.bin_sizes
+        self._cumulative_normed_ps = cumulative_normed_ps
 
-        if cumulative_normed_ps is not None:
-            self.cumulative_normed_ps = cumulative_normed_ps
+    @cached_property
+    def bin_probs(self):
+        return self.normed_densities * constants.bin_sizes
 
-        else:
-            self.cumulative_normed_ps = init_np.append(
-                init_np.array([0]), init_np.cumsum(self.bin_probs)
+    @cached_property
+    def normed_log_densities(self):
+        return np.log(self.normed_densities)
+
+    @cached_property
```

Comment:
Why is this a cached property? Seems like it's doing its own caching?

Reply:
are you referring to normed_log_densities? I'm sorry I don't follow

Reply:
I meant `cumulative_normed_ps`